EIPS/eip-20-token-standard.md

```diff
@@ -0,0 +1,184 @@
+
+
+#### name
+
+Returns the name of the token - e.g. `"MyToken"`.
+
+OPTIONAL - This method can be used to improve useability,
+but interfaces and other contracts MUST NOT expect these values to be present.
+
+
+``` js
+function name() constant returns (string name)
+```
+
+
+#### symbol
+
+Returns the symbol of the token. E.g. "HIX".
+
+OPTIONAL - This method can be used to improve useability,
+but interfaces and other contracts MUST NOT expect these values to be present.
+
+``` js
+function symbol() constant returns (string symbol)
+```
+
+
+
+#### decimals
+
+Returns the number of decimals the token uses - e.g. `8`, means to divide the token amount by `100000000` to get its user representation.
+
+OPTIONAL - This method can be used to improve useability,
+but interfaces and other contracts MUST NOT expect these values to be present.
+
+``` js
+function decimals() constant returns (uint8 decimals)
+```
+
+
+#### totalSupply
+
+Returns the total token supply.
+
+``` js
+function totalSupply() constant returns (uint256 totalSupply)
+```
+
+
+
+#### balanceOf
+
+Returns the account balance of another account with address `_owner`.
+
+``` js
+function balanceOf(address _owner) constant returns (uint256 balance)
+```
+
+
+
+#### transfer
+
+Transfers `_value` amount of tokens to address `_to`, and MUST fire the `Transfer` event.
+The function SHOULD `throw` if the `_from` account balance does not have enough tokens to spend.
+
+A token contract which creates new tokens SHOULD trigger a Transfer event with the `_from` address set to `0x0` when tokens are created.
+
+*Note* Transfers of 0 values MUST be treated as normal transfers and fire the `Transfer` event.
+
+``` js
+function transfer(address _to, uint256 _value) returns (bool success)
+```
+
+
+
+#### transferFrom
+
+Transfers `_value` amount of tokens from address `_from` to address `_to`, and MUST fire the `Transfer` event.
+
+The `transferFrom` method is used for a withdraw workflow, allowing contracts to transfer tokens on your behalf.
+This can be used for example to allow a contract to transfer tokens on your behalf and/or to charge fees in sub-currencies.
+The function SHOULD `throw` unless the `_from` account has deliberately authorized the sender of the message via some mechanism.
+
+*Note* Transfers of 0 values MUST be treated as normal transfers and fire the `Transfer` event.
+
+``` js
+function transferFrom(address _from, address _to, uint256 _value) returns (bool success)
+```
+
+
+
+#### approve
+
+Allows `_spender` to withdraw from your account multiple times, up to the `_value` amount. If this function is called again it overwrites the current allowance with `_value`.
+
+**NOTE**: To prevent attack vectors like the one [described here](https://docs.google.com/document/d/1YLPtQxZu1UAvO9cZ1O2RPXBbT0mooh4DYKjA_jp-RLM/) and discussed [here](https://github.com/ethereum/EIPs/issues/20#issuecomment-263524729),
```

Comment:
it's a pity that `approve` sets the approval amount and doesn't add or remove to it, it was implemented without much thought but now became a standard already.

Reply:
We can change that in another standard, but this one needs to be formalised